language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/imports/AddModuleToBallerinaTomlCodeAction.java

```diff
@@ -0,0 +1,168 @@
+import org.ballerinalang.langserver.codeaction.CodeActionUtil;
+import org.ballerinalang.langserver.common.constants.CommandConstants;
+import org.ballerinalang.langserver.commons.CodeActionContext;
+import org.ballerinalang.langserver.commons.codeaction.spi.DiagBasedPositionDetails;
+import org.ballerinalang.langserver.commons.codeaction.spi.DiagnosticBasedCodeActionProvider;
+import org.ballerinalang.langserver.commons.toml.common.TomlSyntaxTreeUtil;
+import org.ballerinalang.util.diagnostic.DiagnosticErrorCode;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.CodeActionKind;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.TextEdit;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Code Action for add dependency for Ballerina.toml.
+ *
+ * @since 2201.8.0
+ */
+@JavaSPIService("org.ballerinalang.langserver.commons.codeaction.spi.LSCodeActionProvider")
+public class AddModuleToBallerinaTomlCodeAction implements DiagnosticBasedCodeActionProvider {
+
+    public static final String NAME = "Add Module to Ballerina.toml";
+    
+    @Override
+    public boolean validate(Diagnostic diagnostic, DiagBasedPositionDetails positionDetails,
+                            CodeActionContext context) {
+        return CodeActionNodeValidator.validate(context.nodeAtRange());
+    }
+
+    @Override
+    public List<CodeAction> getCodeActions(Diagnostic diagnostic,
+                                           DiagBasedPositionDetails positionDetails,
+                                           CodeActionContext context) {
+        if (!diagnostic.diagnosticInfo().code().equals(DiagnosticErrorCode.MODULE_NOT_FOUND.diagnosticId())) {
+            return Collections.emptyList();
+        }
+        String commandTitle = CommandConstants.ADD_MODULE_TO_BALLERINA_TOML;
+        Optional<Project> project = context.workspace().project(context.filePath());
+        if (project.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        List<DiagnosticProperty<?>> properties = diagnostic.properties();
+        if (properties.isEmpty()) {
+            return Collections.emptyList();
+        }
+        
+        String orgName;
+        String pkgName;
+        String[] orgNameAndRest = ((String) properties.get(0).value()).split("/");
+        if (orgNameAndRest.length == 2) {
+            orgName = orgNameAndRest[0];
+            String[] pkgNameAndRest = orgNameAndRest[1].split("\\.");
+            if (pkgNameAndRest.length == 0) {
+                return Collections.emptyList();
+            }
+            pkgName = pkgNameAndRest[0];
+        } else {
+            return Collections.emptyList();
+        }
+        
+        ArrayList<PackageVersion> versions = new ArrayList<>();
+        getVersions(LSPackageLoader.getInstance(context.languageServercontext()), project.get(), orgName, pkgName, 
```

Comment:
Shall we assign this directly to above variable instead of passing it as an argument